industrial_ci/src/tests/source_tests.sh

```diff
@@ -97,6 +97,96 @@ function run_clang_tidy_check {
     fi
 }
 
+function ici_combine_cpp_reports {
+  ici_install_pkgs_for_command lcov lcov
+  local target_ws=$1
+  cd "$target_ws" || return 1
```

Comment:
if you really want to test it here (I don't see the need), use `ici_error` instead.

Reply:
The scrips must not change the directory.

Reply:
I have put them in subshells, so that they won't change directories, like this. https://github.com/ros-industrial/industrial_ci/blob/5b0bbc77d1bc72ed847974b09bc191f0d38f4de6/industrial_ci/src/tests/source_tests.sh#L147-L149
this
https://github.com/ros-industrial/industrial_ci/blob/5b0bbc77d1bc72ed847974b09bc191f0d38f4de6/industrial_ci/src/tests/source_tests.sh#L167-L171
and this
https://github.com/ros-industrial/industrial_ci/blob/5b0bbc77d1bc72ed847974b09bc191f0d38f4de6/industrial_ci/src/tests/source_tests.sh#L174-L176
Unfortunately, I still cannot completely avoid changing directories, due to the limitation of each command